tests/project_directory_test.php

```diff
@@ -107,4 +107,20 @@ public function testGetDirectoryAvoidPluginsDirectory()
         $this->assertInstanceOf('\\Dxw\\Whippet\\ProjectDirectory', $result->unwrap());
         $this->assertEquals($dir, $result->unwrap()->__toString());
     }
+
+    public function testGetDirectoryWithEnv()
+    {
+        $dir = $this->getDir();
+
+        putenv('WHIPPET_PROJECT_DIRECTORY=/etc');
```

Comment:
The ProjectDirectory class doesn't modify any files. And the `mkdir` and `touch` calls below are operating on files within `vfs://root`. PHP has (partial) support for virtual filesystems. In this case we're using [vfsStream](https://github.com/mikey179/vfsStream) which is an in-memory filesystem meant to be used with tests.